timesketch/api/v1/resources/sigma.py

```diff
@@ -89,4 +94,77 @@ def get(self, rule_uuid):
             abort(
                 HTTP_STATUS_CODE_NOT_FOUND, 'No sigma rule found with this ID.')
 
-        return return_rule
+        meta = {'current_user': current_user.username,
+                'rules_count': len(sigma_rules)}
+        return jsonify({'objects': return_rule, 'meta': meta})
+
+
+class SigmaByTextResource(resources.ResourceMixin, Resource):
+    """Resource to get a Sigma rule by text."""
+
+    @login_required
+    def post(self):
+        """Handles POST request to the resource.
+
+        Returns:
+            JSON sigma rule
+        """
+
+        form = request.json
+        if not form:
+            form = request.data
+
+        action = form.get('action', '')
+
+        # TODO: that can eventually go away since there are no other actions
+        if action != 'post':
+            return abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                'Action needs to be "post"')
+
+        content = form.get('content')
+        if not content:
+            return abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                'Missing values from the request.')
+
+        try:
+            sigma_rule = ts_sigma_lib.get_sigma_rule_by_text(content)
+
+        except ValueError:
+            logger.error('Sigma Parsing error with the user provided rule',
+                         exc_info=True)
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                'Error unable to parse the provided Sigma rule')
+
+        except NotImplementedError as exception:
+            logger.error(
+                'Sigma Parsing error: Feature in the rule provided is not implemented in this backend', exc_info=True) # pylint: disable=line-too-long
```

Comment:
why disable the line too long, wh not just fix that????

```python
logger.error(
    'Sigma parsing error: Feature in the rule provided '
    'is not implemented in this backend.', exc_info=True)
```

and then you don't need to have a too long of a line and don't need to suppress the error message from the linter

Reply:
TIL